langchain4j-oracle/src/main/java/dev/langchain4j/store/embedding/oracle/IndexBuilder.java

```
@@ -0,0 +1,126 @@
+  protected String indexName;
+
+  /**
+   * CreateOption for the index. By default, the index will not be created.
+   */
+  CreateOption createOption = CreateOption.DO_NOT_CREATE;
+
+  /**
+   * Configures the option to create (or not create) an index. The default is
+   * {@link CreateOption#CREATE_IF_NOT_EXISTS}, which means that an index will
+   * be created if an index with the same name does not already exist.
+   *
+   * @param createOption The create option.
+   *
+   * @return This builder.
+   *
+   * @throws IllegalArgumentException If createOption is null.
+   */
+  public T createOption(CreateOption createOption) {
+    ensureNotNull(createOption, "createOption");
+    this.createOption = createOption;
+    return (T) this;
+  }
+
+  /**
+   * Sets the index name.
+   * @param indexName The name of the index.
+   * @return This builder.
+   */
+  public T name(String indexName) {
+    this.indexName = indexName;
+    return (T) this;
+  }
+
+  /**
+   * Creates an index name given the table name and a suffix.
+   * @param tableName The table name.
+   * @param suffix The index suffix.
+   * @return The index name.
+   */
+  String buildIndexName(String tableName, String suffix) {
+    boolean isQuoted =  tableName.startsWith("\"") && tableName.endsWith("\"");
+    // If the table name is a quoted identifier, then the index name must also be quoted.
+    if (isQuoted) {
+      tableName = unquoteTableName(tableName);
+    }
+    indexName = truncateIndexName(tableName + suffix, isQuoted);
+    if (isQuoted) {
+      indexName = "\"" + indexName + "\"";
+    }
+    return indexName;
+  }
+
+  /**
+   * Truncates the index name if it os longer that the maximum length allowed
+   * by the database.
+   * @param indexName The index name.
+   * @param isQuoted True if the index name is quoted.
+   * @return The index name truncated to the max length allowed by the database.
+   */
+  private String truncateIndexName(String indexName, boolean isQuoted) {
```

What if we let the database raise an error in this case?
This way the user would know that their name is too long, and they can change it to a shorter one. My thought is that a user might expect the index to have the exact name they configured. If we use a different name, then the user won't know what to look for when using the index in another SQL command, like a DROP or a REBUILD.
Also, if a future database version supports longer names, then we won't be getting in the way of that.

---

This method is only called when the user does not set the index and a name is generated. I don't think it is a good idea to generate a name that causes an error, that's why I truncate the name. I think we should keep it.

---

Oh, I see. Makes total sense!